autoray/lazy/core.py

```diff
@@ -1267,6 +1267,43 @@ def where(condition, x, y):
     )
 
 
+def indices_shape_for_lazy(indices):
+    shape_list = []
+
+    def helper(sequence):
+        if isinstance(sequence, (list, tuple)):
+            if all(isinstance(x, int) for x in sequence):
+                shape_list.append(len(sequence))
+            elif all((isinstance(x, (list, tuple)) and len(x) == len(sequence[0])) for x in sequence):
+                shape_list.append(len(sequence))
+                helper(sequence[0])
+            else:
+                raise ValueError(
+                    'setting an array element with a sequence. '
+                )
```

Comment:
Some comments might be helpful here as recursive logic is always a little hard to follow.

Reply:
Sure, I will add the comment soon.

Reply:
Instead of using the recursive function, I used a new method to get the shape of indices, it is much more simple.